Tests/FluentAssertions.Json.Specs/JTokenAssertionsSpecs.cs

```diff
@@ -445,6 +445,59 @@ public void When_a_token_is_equal_to_its_string_representation_the_non_equivalen
                 .WithMessage("Expected JSON document not to be equivalent*");
         }
 
+        [Fact]
+        public void When_a_float_is_within_approximation_check_should_succeed()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var actual = JToken.Parse("{ \"id\": 1.1232 }");
+            var expected = JToken.Parse("{ \"id\": 1.1235 }");
+            const double precision = 1e-3;
+            //-----------------------------------------------------------------------------------------------------------
+            // Act & Assert
+            //-----------------------------------------------------------------------------------------------------------
+            actual.Should().BeEquivalentTo(expected,
+                options => options
+                .Using<double>(d => d.Subject.Should().BeApproximately(d.Expectation, precision))
+                .WhenTypeIs<double>());           
+        }
+        [Fact]
+        public void When_a_float_is_not_within_approximation_check_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var actual = JToken.Parse("{ \"id\": 1.1232 }");
+            var expected = JToken.Parse("{ \"id\": 1.1235 }");
+            const double precision = 1e-5;
+            //-----------------------------------------------------------------------------------------------------------
+            // Act & Assert
+            //-----------------------------------------------------------------------------------------------------------
+            
+            actual.Should().
+                Invoking(x => x.BeEquivalentTo(expected,options => options
+                .Using<double>(d => d.Subject.Should().BeApproximately(d.Expectation, precision))
+                .WhenTypeIs<double>()))
+                .Should().Throw<XunitException>()
+                .WithMessage("JSON document has a different value at $.id.*");
+        }
+        [Fact]
+        public void When_default_equivalencyassertionoptions_used_check_shoud_not_change_it()
```

Comment:
As `AssertionOptions.AssertEquivalencyUsing` modifies the global options, we need to shield other test from applying the same options.
See [`When_assertion_doubles_should_always_allow_small_deviations`](https://github.com/fluentassertions/fluentassertions/blob/8450e98d0ee5c25f28249b0efd99b82b68e5ead6/Tests/FluentAssertions.Specs/AssertionOptionsSpecs.cs#L135) for an example.